Auth/Support/Oauth2/Token/IntegrationTokenFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * @copyright   2019 Mautic, Inc. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.com
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace MauticPlugin\IntegrationsBundle\Auth\Support\Oauth2\Token;
+
+use kamermans\OAuth2\Token\TokenInterface;
+
+class IntegrationTokenFactory implements TokenFactoryInterface
+{
+    private $extraKeysToStore = [];
+
+    public function __construct(array $extraKeysToStore = [])
+    {
+        $this->extraKeysToStore = $extraKeysToStore;
+    }
+
+    public function __invoke(array $data, TokenInterface $previousToken = null)
```

Comment:
I didn't fully grasp what's the perk for using this magic method over normal public method. Is there a reason for it?
```suggestion
    public function __invoke(array $data, TokenInterface $previousToken = null): IntegrationToken
```

Reply:
@escopecz The vendor uses the factory in its code as an invokable and thus the requirement here. I don't know why nor why they didn't provide an interface but we don't have control over it.